documentation/specs/proposed/decoupling-vs-sdk.md

```diff
@@ -0,0 +1,43 @@
+# Decoupling VS builds of SDK projects
+The behaviour of an SDK project, when built in Visual Studio and when built in the DotNet CLI can vary, as different pieces of imports can be fetched from either build agent. This situation is not great specially for some developers as it increases their coding workload. To solve this, we want to ensure that all logic from a build from an *SDK project* comes from the SDK, independently of where the build is processed.
+
+## Goals and Motivation
+
+We are aiming for:
+ - Consistent end-user eperience for build in either DotNet CLI or Visual Studio.
+ - Isolating Dotnet SDK build to only their components.
+ - Decoupling the Dotnet SDK from VS.
+
+
+The reason we are persuing this is for a better experience when using or writting Roslyn analyzers and MSBuild Tasks. Currently tooling authors need to target NetStandard2.0 for their projects to be recognized by VS. Another options is to multi-target but it takes a lot more effort and time spent on that. Another aspect is the user experience, if the Roslyn version for VS analyzers and generators doesn't match the one in Visual Studio, they do not work.
+
+
+## Impact
+There are a few area of impact:
+ - SDK style project builds will be more stable between VS and CLI builds, as the tooling will not be devided between different versions.
+ - Reduced cost of development for external and internal teams that contribute to Roslyn Analyzers, SourceBuild generators, or MSBuild Tasks.
+ - End-user will not experience mismatch between analyzer versions, and confirmation that their SDK style builds will behave the same way in VS and in the command line.
+
+## Stakeholders
+For the internal stakeholder, we have the teams that will continue the work to fully complete the VS and SDK decoupling feature after our base work is done. There are two handovers in this project:
+
+1. Enabling the MSBuild.exe execution state to be the same as DotNet command line invocation so the Roslyn team can enable the use of their Core compiler in VS.
+2. Tasks and other projects can be written in .NET core and the SDK projects will build successuflly in VS. This enables other teams to migrate their tasks to .NET core instead of keeping them targeting .NET Framework.
+
+The handovers should allow other teams to proceed with their work smoothly and no change in build behaviour should be present within MSBuild.
+
+## Risks
+A few risks associated with this feature:
+ - There might be a performance hit on VS. It would depending on the amount of non-framwork tasks that the project will need to load when opening it in VS. The performance gain from pre-loading will not be available in this scenario.
+ - The VS tooling might present have a different version than the SDK installed, which might lead to discrepancy in partial builds.
```

Comment:
Part of our statement now is that VS and .NET SDK are de-coupled experiences so I wouldn't put this in the risk section. This is more leaning into our design of the system. I would put this in the impact section and say that an impcat of this change is it re-enforces our design that VS and .NET SDK be decoupled experiences.

Reply:
When you put it that way it sounds like a consequence of the design, which is not really covered in this doc specific. So I opted to take it out as a risk in this case.